src/app.js

```diff
@@ -81,8 +87,13 @@ export function app(app) {
             }).data
           )
 
-          if (result != null && result.then == null) {
-            repaint((state = merge(state, emit("update", result))))
+          if (result == null) {
+          } else if (typeof result == "function") {
+            result = result(update)
+          } else if (typeof result.then == "function") {
+            result.then(update)
+          } else {
```

Comment:
Could we make this `else if (result != null)` and then remove then empty `if (result == null) {}` above?